### PR TITLE
hwmv2: STM32: Fixes

### PR DESCRIPTION
**Pull request by ajarmouni-st**
Cleanup & fix for CI issue https://github.com/zephyrproject-rtos/zephyr/pull/68801#issuecomment-1935986243

### CODE REVIEW

File: boards/st/stm32f3_disco/stm32f3_disco_stm32f303xc_E.overlay
Comment (erwango) [inline boards/st/stm32f3_disco/stm32f3_disco_stm32f303xc_E.overlay:1]: Shouldn't we update .conf and .yaml files as well ?

File: boards/st/nucleo_f030r8/nucleo_f030r8_1.overlay
Comment (erwango) [inline boards/st/nucleo_f030r8/nucleo_f030r8_1.overlay:1]: Should be in previous commit

File: boards/st/stm32f3_disco/stm32f3_disco_E.overlay
Comment (erwango) [inline boards/st/stm32f3_disco/stm32f3_disco_E.overlay:1]: Should be in previous commit

**Review by erwango — CHANGES_REQUESTED** [review]

File: boards/st/stm32f3_disco/stm32f3_disco_stm32f303xc_E.overlay
Comment (ajarmouni-st) [reply boards/st/stm32f3_disco/stm32f3_disco_stm32f303xc_E.overlay:1]: I don't think so

**Review by erwango — APPROVED** [review]

**Review by nordicjm — APPROVED** [review]

**Review by nordicjm — CHANGES_REQUESTED** [review]
> ERROR   - tests/drivers/clock_control/stm32_clock_configuration/stm32_common_devices/drivers.clock.stm32_clock_configuration.common_device.f3.i2c1_hsi - platform_allow - unrecognized platform - stm32f3_disco

**Review by nordicjm — APPROVED** [review]

**Pull request merged**